lib/sampler.js

```diff
@@ -206,15 +206,15 @@ var sampler = (module.exports = {
       logger.debug('Not adding CPU metric sampler.')
     }
 
-    sampler.state = 'running'
+    this.state = 'running'
```

Comment:
Just calling out that the change to using 'this' directly means a refactor could break the code by changing what 'this' is. Pretty low risk, given this is used only by us and a standard risk in the JavaScript exports we commonly use.

Reply:
and there are tests for asserting all these cases now so we should be fine. this one was missing so I added a test